operate/schema/src/test/java/io/camunda/operate/schema/IndexSchemaValidatorIT.java

```diff
@@ -0,0 +1,268 @@
+ * "Non-production use" means any use of the Software that is not directly related to creating products, services, or systems that generate revenue or other direct or indirect economic benefits.  Examples of permitted non-production use include personal use, educational use, research, and development. Examples of prohibited production use include, without limitation, use for commercial, for-profit, or publicly accessible systems or use for commercial or revenue-generating purposes.
+ *
+ * If the Licensee is in breach of the Conditions, this Agreement, including the rights granted under it, will automatically terminate with immediate effect.
+ *
+ * SUBJECT AS SET OUT BELOW, THE SOFTWARE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE, AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ * NOTHING IN THIS AGREEMENT EXCLUDES OR RESTRICTS A PARTY’S LIABILITY FOR (A) DEATH OR PERSONAL INJURY CAUSED BY THAT PARTY’S NEGLIGENCE, (B) FRAUD, OR (C) ANY OTHER LIABILITY TO THE EXTENT THAT IT CANNOT BE LAWFULLY EXCLUDED OR RESTRICTED.
+ */
+package io.camunda.operate.schema;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.assertj.core.api.Assertions.entry;
+
+import io.camunda.operate.JacksonConfig;
+import io.camunda.operate.connect.ElasticsearchConnector;
+import io.camunda.operate.connect.OpensearchConnector;
+import io.camunda.operate.connect.OperateDateTimeFormatter;
+import io.camunda.operate.exceptions.OperateRuntimeException;
+import io.camunda.operate.schema.IndexMapping.IndexMappingProperty;
+import io.camunda.operate.schema.elasticsearch.ElasticsearchSchemaManager;
+import io.camunda.operate.schema.indices.IndexDescriptor;
+import io.camunda.operate.schema.opensearch.OpensearchSchemaManager;
+import io.camunda.operate.schema.util.SchemaTestHelper;
+import io.camunda.operate.schema.util.TestIndex;
+import io.camunda.operate.schema.util.TestTemplate;
+import io.camunda.operate.schema.util.elasticsearch.ElasticsearchSchemaTestHelper;
+import io.camunda.operate.schema.util.opensearch.OpenSearchSchemaTestHelper;
+import io.camunda.operate.store.elasticsearch.ElasticsearchTaskStore;
+import io.camunda.operate.store.opensearch.OpensearchTaskStore;
+import java.util.Map;
+import java.util.Set;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.ContextConfiguration;
+
+@ContextConfiguration(
+    classes = {
+      IndexSchemaValidator.class,
```

Comment:
I have concerns about making ITs like this that only load certain classes. In addition to the difficulty of maintaining (as we add new dependencies and components, these types of tests randomly break and need the class list expanded), it only causes a small part of the application to be loaded and can miss issues. Is there an issue with loading the application with the same full context as our other ITs?

Reply:
I think I started initially with the full application context approach and it became a bit frustrating, but I don't remember the details. I think problems were that there was no ready-made Spring Boot application (in the schema module) that creates the full context for the schema. And using package scanning, it was a situation that one thing only led to the next, with more and more beans unrelated to the test getting added in, requiring more configuration and doing more unrelated `@PostConstruct` logic.

That said, I also agree with your points. The current design was the in my opinion better tradeoff of having focused tests in the module at the expense of maintainability. Here as well, you could go in either direction.

Reply:
If it's too complex to run with a full application context, then this approach makes sense. I just wanted to raise the issue because especially for tests that are not run as part of the main pipeline (performance, migration, etc), if they break because of a new dependency added, it might not get caught for a while. We had situations where tests like these were found to be broken during a release just because of some changed dependencies.